### ISSUE TITLE
Subsets of geno files

### ISSUE THREAD

**Issue by axelwilhelm** [ANGSD/angsd#513]
Hi all
Maybe this does not belong here but I just realised that I am always creating new geno files every time I do some analysis.
I was thinking that if I have a large geno file (a chromosome) I would be able to extract regions from it in further analyses (ngsTools).
Seems like an obvious issue but then it might be indicating lack of clarity in the documentation/tutorials (or just me). 
How are others using geno files? I must be doing it incorrectly.

**Comment by TeresaPegan** [comment]
Hi, 
I make a large file of genotype likelihoods -- sometimes even of the entire genome -- and then I just use awk and sed and other text manipulation code to extract regions I want to focus on. Ultimately the genotype likelihood files are just text files and can be manipulated accordingly. I agree that the documentation of these programs makes it seems like the default is to make a new file every time, which is a waste of computation.
-Teresa

**Comment by axelwilhelm** [comment]
Now if I 'hexdump' a gzipped geno file I see this:

0000000 6e34 0a45 6c2f 3fda 48e6 7add c9e8 3fe2
0000010 0539 f5f5 036f 3c05 7e55 72e0 5063 3bd1
0000020 f7ba ffff ffff 3fef eda8 5cc9 8b3e 3d50
...etc....

And I have no idea how I would awk this for my needs

**Comment by TeresaPegan** [comment]
I usually use gzipped beagle files, so if you are using binary files I am not quite sure if you would need a different approach. When things are gzipped, they need to be uncompressed before you can manipulate them. You can do this nicely with zcat and piping. For example, this code takes a genome-wide beagle file and appends the chromosome label (without the _position part -- this only works if there is no underscore in the chromosome name) as a new first column. Later I could filter the beagle file based on the chromosome name if I wanted to pull out values associated with a chromosome of interest. 
```
zcat $BEAGLE | awk -F'_' -v OFS='\t' '{print $1,$0}' 
```

Unfortunately it can take quite a while to learn how to manage these files with awk, but there is no way to subset beagle files within the ngsTools programs, as far as I know.
-Teresa

**Comment by axelwilhelm** [comment]
Yes I was aware of the beagle option and have used it. I am just stuck in the comfort of old scripts that used the geno file format... so stuck that I refused to consider the beagle option... there was some reason for it at first that I don't even remember anymore.
I'll just switch to beagle, unless I remember why I didn't in the first place
